packages/next/src/server/after/after-context.ts

```diff
@@ -109,11 +109,19 @@ export class AfterContext {
 
     const staticGenerationStore = staticGenerationAsyncStorage.getStore()
 
-    return withExecuteRevalidates(staticGenerationStore, () =>
-      requestAsyncStorage.run(readonlyRequestStore, async () => {
-        this.callbackQueue.start()
-        await this.callbackQueue.onIdle()
-      })
+    return staticGenerationAsyncStorage.run(
+      {
+        ...staticGenerationStore!,
```

Comment:
We do this cloning elsewhere in unstable_cache but I’m pretty sure this leads to bugs because we mutate the original object sometimes which gets shadowed. Eg when we add the revalidate promises that mutates from undefined to array.

So we can’t shadow it this way.

Reply:
I think it might be ok to just mutate it instead.

Reply:
the `withExecuteRevalidates` function used below has some special logic around this that might make this okay
https://github.com/ijjk/next.js/blob/36097226ff5a9a3c671de8633e899ffd310e69ae/packages/next/src/server/after/revalidation-utils.ts#L11-L13
...unless that's buggy too?

(see PR that introduced it: https://github.com/vercel/next.js/pull/70458)

Reply:
Updated to mutate instead as that sounds safe for this case.